dimagi/utils/web.py

```diff
@@ -16,18 +20,19 @@
 from datetime import date, datetime, time
 from decimal import Decimal
 
+
 def get_url_base():
+    return '{}://{}'.format(settings.DEFAULT_PROTOCOL, get_site_domain())
+
+
+def get_site_domain():
     try:
-        protocol = settings.DEFAULT_PROTOCOL
+        return settings.BASE_ADDRESS
     except AttributeError:
-        protocol = 'http'
-    address = getattr(settings, "BASE_ADDRESS", None)
-    if address is None:
-        address = Site.objects.get(id = settings.SITE_ID).domain
-    return '%s://%s' % (protocol, address)
-
-def get_secure_url_base():
-    return 'https://%s' % Site.objects.get(id = settings.SITE_ID).domain
+        warnings.warn("Please don't use Site to set domain anymore; "
+                      "use BASE_ADDRESS instead.", DeprecationWarning)
+        return Site.objects.get(id=settings.SITE_ID).domain
```

Comment:
I'm not sure what you mean. I was thinking that if I get rid of all references to `Site` in this fashion throughout the codebase, and keep this here as a failover that displays a deprecation warning, people would be able to fix this for themselves as they notice it. We already basically require `BASE_ADDRESS` for a bunch of other things, and for that to be identical to what was in the `Site` model, so likely most people already have it.